### ISSUE TITLE
Rectify the fortran code generation for `maskr`/`maskl` intrinsic 

### ISSUE THREAD

**Issue by HarshitaKalani** [lfortran/lfortran#4082]
```fortran
 result = maskr(63, 8)
```
The above fortran program is converted to the below program with asr_to_fortran. 
```fortran
result = maskr(63)
```
We need to fix this.

**Comment by kmr-srbh** [comment]
This is an issue with code-generation for other intrinsics also.
```fortran
program example
   print*, char(65, 1)
end
```
```console
(base) saurabh-kumar@Awadh:~/Projects/System/lfortran$ lfortran ./examples/example.f90 --show-fortran
program example
implicit none
print *, char(65)
end program example
```

**Comment by kmr-srbh** [comment]
Checking the ASR, we find that the argument is not being passed to the list of arguments. This issue is related to only the `KIND` argument. We handle it specially for the return type and do not add it to the list of arguments. Other intrinsics like `max` which do not have `KIND` as an argument, do not give a wrong output.

```fortran
program example
   print*, max(1, 2, 3, 4, 5)
end
```
```console
(base) saurabh-kumar@Awadh:~/Projects/System/lfortran$ lfortran ./examples/example.f90 --show-fortran
program example
implicit none
print *, max(1, 2, 3, 4, 5)
end program example
```

**Comment by HarshitaKalani** [comment]
Yes, we just consider `x.n_args` number of args in asr_to_fortran and the optional kind arg is not included in that.

**Comment by Pranavchiku** [comment]
This is how it is designed @kmr-srbh thing is we need kind only to set return type and while `create_XX` we use it and discard information because if we keep it, that will be redundant. So, if to fix this issue you need to register intrinsic that supports optional argument kind, fetch return type and set `kind = ASRUtils::extract_kind_from_ttype_t(return_type)`.
